src/test/kotlin/open/tresorier/services/OperationServiceTest.kt

```diff
@@ -63,4 +63,87 @@ class OperationServiceTest : ITest {
+    }
+
+    @Test
+    fun testOfxFileWithoutOperation() {
+        val emmanuelle: Person = personService.createPerson(
+            "Emmanuelle Carpentier", "Nobel_2020", "emmanuelle@chimie.fr", ProfileEnum.PROFILE_USER
+        )
+        emmanuelle.billingStatus = true
+        val budget: Budget = budgetService.findByUser(emmanuelle)[0]
+        val account: Account = accountService.create(
+            emmanuelle, budget, "personal account", TestData.jan_14_2022, 1000
+        )
+        val ofxFile: String ="<OFXHEADER:100DATA:OFXSGMLVERSION:102SECURITY:NONEENCODING:USASCIICHARSET:1252COMPRESSION:NONEOLDFILEUID:NONENEWFILEUID:NONE<OFX><SIGNONMSGSRSV1><SONRS><STATUS><CODE>0<SEVERITY>INFO</STATUS><DTSERVER>20220302<LANGUAGE>FRA<DTPROFUP>20220302<DTACCTUP>20220302</SONRS></SIGNONMSGSRSV1><BANKMSGSRSV1><STMTTRNRS><TRNUID>00<STATUS><CODE>0<SEVERITY>INFO</STATUS><STMTRS><CURDEF>EUR<BANKACCTFROM><BANKID>42559<BRANCHID>10000<ACCTID>04065601113<ACCTTYPE>SAVINGS</BANKACCTFROM><BANKTRANLIST><DTSTART>20220215<DTEND>20220228</BANKTRANLIST><LEDGERBAL><BALAMT>5693,5<DTASOF>20220302</LEDGERBAL></STMTRS></STMTTRNRS></BANKMSGSRSV1></OFX>"
+        val listOperations: List<String> = operationService.separateOperationsOfOfxFile(emmanuelle, account, ofxFile)
+
+        Assertions.assertEquals(listOperations.size, 0)
+    }
+
+    @Test
+    fun testOfxFileWithOnlyOneOperation() {
+        val katherine: Person = personService.createPerson(
+            "Katherine Johnson", "NASA_1953", "katherine@fusée.com", ProfileEnum.PROFILE_USER
+        )
+        katherine.billingStatus = true
+        val budget: Budget = budgetService.findByUser(katherine)[0]
+        val account: Account = accountService.create(
+            katherine, budget, "personal account", TestData.jan_14_2022, 1000
+        )
+        val ofxFile: String ="<OFXHEADER:100DATA:OFXSGMLVERSION:102SECURITY:NONEENCODING:USASCIICHARSET:1252COMPRESSION:NONEOLDFILEUID:NONENEWFILEUID:NONE<OFX><SIGNONMSGSRSV1><SONRS><STATUS><CODE>0<SEVERITY>INFO</STATUS><DTSERVER>20220302<LANGUAGE>FRA<DTPROFUP>20220302<DTACCTUP>20220302</SONRS></SIGNONMSGSRSV1><BANKMSGSRSV1><STMTTRNRS><TRNUID>00<STATUS><CODE>0<SEVERITY>INFO</STATUS><STMTRS><CURDEF>EUR<BANKACCTFROM><BANKID>42559<BRANCHID>10000<ACCTID>04065601113<ACCTTYPE>SAVINGS</BANKACCTFROM><BANKTRANLIST><DTSTART>20220215<DTEND>20220228<STMTTRN><TRNTYPE>DEBIT<DTPOSTED>20220228<TRNAMT>-8,85<FITID>2802202220220228-17.02.34.872431<NAME>MERCURY REDSTONE 3<MEMO>MERCURY REDSTONE 3</STMTTRN></BANKTRANLIST><LEDGERBAL><BALAMT>5693,5<DTASOF>20220302</LEDGERBAL></STMTRS></STMTTRNRS></BANKMSGSRSV1></OFX>"
+        val listOperations: List<String> = operationService.separateOperationsOfOfxFile(katherine, account, ofxFile)
+
+        Assertions.assertTrue(listOperations.size == 1 && listOperations[0] == "<STMTTRN><TRNTYPE>DEBIT<DTPOSTED>20220228<TRNAMT>-8,85<FITID>2802202220220228-17.02.34.872431<NAME>MERCURY REDSTONE 3<MEMO>MERCURY REDSTONE 3</STMTTRN>")
+
+    }
```

Comment:
more test : what happen if the date is invalid ? ex 20221521. there is no 15th month
what happen if a section is missing (no amount or no date ?)
=> I think an exception should be thrown then ignored (so it can be logged but without interrupting the process of all the other import)

I think it could be useful to return to the front the number of imported operations and the number of fail if there was any (maybe with the ofxOperation version of the invalid operation, so that an advanced user could look at it and manually add the corresponding operation without having to look for hte missing operation line by line).